models/topics.model.js

```diff
@@ -1,35 +1,67 @@
-const db = require('../db/connection')
-const fs = require('fs/promises')
+const db = require('../db/connection');
+const fs = require('fs/promises');
 
-function findALlTopics(){
-    return db
-    .query('SELECT * FROM topics')
-    .then((result => {
-        return result.rows
-    }))
+function findALlTopics() {
+  return db.query('SELECT * FROM topics').then((result) => {
+    return result.rows;
+  });
 }
 
-function findArticleById(articleId){
-    return db.query(`SELECT * FROM articles WHERE article_id = $1`, [articleId])
+function findArticleById(articleId) {
+  return db
+    .query(`SELECT * FROM articles WHERE article_id = $1`, [articleId])
     .then((result) => {
-        return result.rows
-    })
+      return result.rows;
+    });
 }
 
-function findAllArticles(){
-    // const commentCount = db.query(`SELECT COUNT(body) AS comment_count FROM comments WHERE `)
-    return db.query(`SELECT articles.*, COUNT(comments.comment_id) AS comment_count
+function findAllArticles() {
+  // const commentCount = db.query(`SELECT COUNT(body) AS comment_count FROM comments WHERE `)
+  return db
+    .query(
+      `SELECT articles.*, COUNT(comments.comment_id) AS comment_count
     FROM articles 
     LEFT JOIN comments ON articles.article_id = comments.article_id 
     GROUP BY articles.article_id
-    ORDER BY articles.created_at DESC`)
+    ORDER BY articles.created_at DESC`
+    )
+    .then((result) => {
+      const editedOutput = result.rows.map((article) => {
+        delete article.body;
+        return article;
+      });
+      return editedOutput;
+    });
+}
+
+function findCommentsByArticle(articleId) {
+  const id = [articleId];
+//   console.log('find comms')
+  return db
+    .query(
+      `SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at DESC`,
+      id
+    )
     .then((result) => {
-        const editedOutput = result.rows.map((article) => {
-            delete article.body
-            return article
-        })
-        return editedOutput
+      return result.rows;
     })
+    .catch((err) => {return Promise.reject({'status':400, 'msg':'No results Found*'})})
```

Comment:
As we should only have one catch block per promise chain catch blocks should only be on the controller (at the end of the chain). If you Promise.reject from the model this rejection will end up in the controller catch block and then you handle it from there.